core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -63,7 +64,8 @@ protected BaseMetadataTable(TableOperations ops, Table table, String name) {
   static PartitionSpec transformSpec(Schema metadataTableSchema, PartitionSpec spec) {
     PartitionSpec.Builder identitySpecBuilder =
         PartitionSpec.builderFor(metadataTableSchema).checkConflicts(false);
-    spec.fields().forEach(pf -> identitySpecBuilder.add(pf.fieldId(), pf.name(), "identity"));
+    spec.fields()
```

Comment:
nit: using a for loop would allow us to format without splitting the statement into multiple lines

Reply:
Fixed.